File: Graph/02_bfs.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // Function to return Breadth First Traversal of given graph.
    
    vector<int> bfsGraph(vector<vector<int>>& adj,vector<int>&visited){
        vector<int> ans;
        queue<int> q;
        q.push(0);
        visited[0]=1;
        while(!q.empty()){
            int node=q.front();
            q.pop();
            ans.push_back(node);
            for(auto it:adj[node]){
                if(visited[it]==0){
                    q.push(it);
                    visited[it]=1;
                }
            }
            
        }
        return ans;
    }
    vector<int> bfs(vector<vector<int>> &adj) {
        // Code here
        
       
        vector<int> visited(adj.size(),0);
        
        return bfsGraph(adj,visited);
        
    }
};

File: Graph/02_bfs.java
class Solution {
    // Function to return Breadth First Search Traversal of given graph.
    
    
    public ArrayList<Integer> bfs(ArrayList<ArrayList<Integer>> adj) {
        // code here
        int n=adj.size();
        boolean[] visited=new boolean[n];
        ArrayList<Integer> ans=new ArrayList<Integer>();
        Queue<Integer> q=new LinkedList<>();
        q.offer(0);
        visited[0]=true;
        while(!q.isEmpty()){
            int node=q.poll();
            ans.add(node);
            for(int neighbour:adj.get(node)){
                if(!visited[neighbour]){
                    visited[neighbour]=true;
                    q.offer(neighbour);
                }
            }
        }
        return ans;
    }
}